wmidump.c

```diff
@@ -156,10 +159,16 @@ static void *parse_ascii_wdg(const char *wdg, size_t *bytes)
 			continue;
 		if (!isalnum(*p))
 			continue;
-		char c = strtol(p, &p, 16);
+
+		errno = 0;
+		lval = strtol(p, &p, 16);
+		if (errno == ERANGE && (lval == LONG_MAX || lval == LONG_MIN))
+			errx(1, "<stdin>:%ld:%ld: invalid hexadecimal number",
```

Comment:
While you're checking the range, it seems like it would be worth verifying in lands within the range of the 8bit char which lval gets truncated to in the assignment to the data array. (I can't seem to comment on the individual commit, maybe I'm doing something wrong).